bpf/libbpf/libbpf_api.h

```diff
@@ -26,7 +26,12 @@ static void set_errno(int ret) {
 struct bpf_object* bpf_obj_open(char *filename) {
 	struct bpf_object *obj;
 	obj = bpf_object__open(filename);
-	set_errno(libbpf_get_error(obj));
+	int err = libbpf_get_error(obj);
+	if (err) {
+		free(obj);
```

Comment:
If err is non-zero I think you don't need to free anything because it won't be a valid pointer?